### PR TITLE
stm32-vbat: small improvements to PR#43863

### PR DESCRIPTION
**Pull request by aurel32**
Sorry for not reviewing PR #43863 earlier. I have a few suggestions that I implemented in this PR.

### CODE REVIEW

File: drivers/adc/adc_stm32.c [hunk 1]
```diff
@@ -873,7 +873,7 @@ static void adc_stm32_setup_channels(const struct device *dev, uint8_t channel_i
 		__LL_ADC_CHANNEL_TO_DECIMAL_NB(LL_ADC_CHANNEL_VREFINT) == channel_id) {
 		adc_stm32_set_common_path(dev, LL_ADC_PATH_INTERNAL_VREFINT);
 	}
-#if defined(CONFIG_STM32_VBAT)
+#if defined(LL_ADC_CHANNEL_VBAT)
```
Comment (erwango) [inline drivers/adc/adc_stm32.c:875]: Need to clean up the #endif below.

**Review by erwango — CHANGES_REQUESTED** [review]

File: drivers/sensor/stm32_vbat/stm32_vbat.c [hunk 2]
```diff
@@ -72,7 +71,8 @@ static int stm32_vbat_channel_get(const struct device *dev, enum sensor_channel
 		return -ENOTSUP;
 	}
 
-	voltage = data->raw * cfg->vref_mv / 0x0FFF; /* Sensor value in millivolts */
+	/* Sensor value in millivolts */
+	voltage = data->raw * adc_ref_internal(data->adc) / 0x0FFF;
 	/* considering the vbat input through a resistor bridge */
 	voltage = voltage * cfg->ratio / 1000; /* value of SENSOR_CHAN_VOLTAGE in Volt */
```
Comment (gmarull) [inline drivers/sensor/stm32_vbat/stm32_vbat.c:77]: note: this driver could likely be improved by using adc_dt_spec facilities (out of the scope of this PR)

**Review by gmarull — COMMENTED** [review]
-

File: drivers/adc/adc_stm32.c (hunk 1 above)
Comment (aurel32) [reply drivers/adc/adc_stm32.c:877]: Oops, good catch. I have just fixed that.

**Review by erwango — APPROVED** [review]

File: drivers/sensor/stm32_vbat/stm32_vbat.c (hunk 2 above)
Comment (aurel32) [reply drivers/sensor/stm32_vbat/stm32_vbat.c:77]: Yes, unfortunately this won't work in that case (or rather not without warnings in the DT generation) due to issue #47119.

That said, I have used that for the more generic implementation in PR #47176. Feedback is welcome.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**